### DISCUSSION TITLE
The behaviour of should_auto_renew

### DISCUSSION THREAD

**Discussion by d3v1** [etsy/open-api#690] [Q&A] [answered]
Hey,

Thanks for all the great work on the v3 API.

I have a question about how the `should_auto_renew` property should behave for listings. The API docs state

> When true, renews a listing for four months upon expriation. If set to true when previously false, etsy.com renews the listing before it expires, but the renewal period starts immidiately rather than extending the listing's expiration date. Any unused time remaining on the listing is lost. Renewals result in charges to a user's bill.

My understanding would be that if the value was `false` and then was flipped to `true` it should extend the expiry date by 4 months. The explanation above is not particularly clear though.

I've tested this by:

- Taking an existing listing with the `should_auto_renew` set to `true`
- Flipped `should_auto_renew` to `false`
- Flipped `should_auto_renew` to `true`
- Observed no change in the expiry date

I tested this with a very short time between the flip (seconds), a medium interval (15mins) and a longer interval (16 hours). None of the tests resulted in extending the expiry date.

What is the expected behaviour? It would be great to get a clear explanation and add it to the docs.

Thanks for your help!

**Comment by etsyachristensen** [comment, marked as answer]
@d3v1 Our documentation is actually incorrect here. Turns out this was a conflation from when V2 allowed you do simply "renew" the listing (versus having an auto_renew option). We are correcting the docs.
